src/plugins/discord/fetcher.test.js

```diff
@@ -0,0 +1,299 @@
+      const fetcher = new Fetcher(fetch, options());
+      const {results} = await fetcher.members("1", "0");
+      expect(results).toEqual(response);
+    });
+
+    it("returns correct endCursor", async () => {
+      const response: any[] = [testMember("1"), testMember("2")];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {endCursor} = (await fetcher.members("1", "0")).pageInfo;
+      expect(endCursor).toEqual("2");
+    });
+
+    describe("next page", () => {
+      it("next page = true", async () => {
+        const response: any[] = [testMember("1")];
+        const fetch = jest.fn(() => Promise.resolve(response));
+        const fetcher = new Fetcher(fetch, {
+          membersLimit: 1,
+          messagesLimit: 100,
+          reactionsLimit: 100,
+        });
+        const {hasNextPage} = (await fetcher.members("1", "0")).pageInfo;
+        expect(hasNextPage).toBe(true);
+      });
+
+      it("next page = false", async () => {
+        const response: any[] = [testMember("1")];
+        const fetch = jest.fn(() => Promise.resolve(response));
+        const fetcher = new Fetcher(fetch, {
+          membersLimit: 2,
+          messagesLimit: 100,
+          reactionsLimit: 100,
+        });
+        const {hasNextPage} = (await fetcher.members("1", "0")).pageInfo;
+        expect(hasNextPage).toBe(false);
+      });
+    });
+  });
+
+  describe("fetch reactions", () => {
+    const emoji: Emoji = {id: "1", name: "emojiname"};
+
+    const options = () => ({
+      membersLimit: 100,
+      messagesLimit: 100,
+      reactionsLimit: 2,
+    });
+
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, options());
+      await fetcher.reactions("1", "2", emoji, "0");
+      expect(fetch.mock.calls[0]).toEqual([
+        `/channels/1/messages/2/reactions/emojiname:1?after=0&limit=2`,
+      ]);
+    });
+
+    it("handles response", async () => {
+      // response object from server, prior to type transformation
+      const response: any[] = [{id: "123", emoji}];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {results} = await fetcher.reactions("3", "2", emoji, "0");
+      const expected: Reaction[] = [
+        {emoji, channelId: "3", messageId: "2", authorId: "123"},
+      ];
+      expect(results).toEqual(expected);
+    });
+
+    it("returns correct endCursor", async () => {
+      const response: any[] = [
+        {id: 1, emoji},
+        {id: 2, emoji},
+      ];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {endCursor} = (
+        await fetcher.reactions("1", "2", emoji, "0")
+      ).pageInfo;
+      expect(endCursor).toBe(2);
```

Comment:
This `2` is ambiguous. It could be the "2" argument of the function or `id: "2"` from the API response.
Can you make sure it's a distinct value?

Reply:
Good point, updated